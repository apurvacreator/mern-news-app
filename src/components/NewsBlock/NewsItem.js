import React from 'react';
import PropTypes from 'prop-types';
import { Card, SafeAnchor } from 'react-bootstrap';
import NoImage from '../../no-camera.svg';

const NewsItem = ({ urlToImage, title, url, date }) => (
    <Card style={{ width: '18rem' }}>
        <Card.Img variant="top" src={urlToImage || NoImage} alt={title} />
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Text>{new Date(date).toLocaleDateString()}</Card.Text>
            <SafeAnchor variant="primary" href={url} target='_blank'>Read the article</SafeAnchor>
        </Card.Body>
    </Card>
)

NewsItem.propTypes = {
    urlToImage: PropTypes.string,
    title: PropTypes.string,
    url: PropTypes.string,
    date: PropTypes.string,
}

export default NewsItem;