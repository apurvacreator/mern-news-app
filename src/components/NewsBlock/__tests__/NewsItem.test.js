import React from 'react'
import { mount } from 'enzyme'
import NewsItem from '../NewsItem'

const news = {
    urlToImage: "https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg",
    title: 'This is a test news',
    url: 'https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/',
    date: '2020-08-13T11:46:00Z'
}

describe('NewsItem', () => {
    let wrapper
    beforeEach(() => {
        wrapper = mount(<NewsItem {...news} />)
    })
    it('should render', () => {
        expect(wrapper).toBeTruthy()
    })
    it('should match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})