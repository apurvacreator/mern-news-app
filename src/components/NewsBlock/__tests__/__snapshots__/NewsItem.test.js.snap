// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem should match snapshot 1`] = `"<div style=\\"width: 18rem;\\" class=\\"card\\"><img class=\\"card-img-top\\" src=\\"https://cbsnews3.cbsistatic.com/hub/i/r/2020/08/12/620d4e6e-bcfa-45af-8599-ba9972b6f17d/thumbnail/1200x630/7c2070a64901e2f363b1d75058f86ce8/gettyimages-1180699958.jpg\\" alt=\\"This is a test news\\"><div class=\\"card-body\\"><div class=\\"card-title h5\\">This is a test news</div><p class=\\"card-text\\">8/13/2020</p><a variant=\\"primary\\" href=\\"https://www.cbsnews.com/news/big-12-college-football-2020-season-big-ten-pac-12-postponements/\\" target=\\"_blank\\">Read the article</a></div></div>"`;
